include/attentive/at.h

```diff
@@ -81,6 +81,14 @@ void at_set_callbacks(struct at *at, const struct at_callbacks *cbs, void *arg);
  */
 void at_set_command_scanner(struct at *at, at_line_scanner_t scanner);
 
+/**
+ * Set custom per-character handler for the next command.
+ *
+ * @param at AT channel instance.
+ * @param handler Character handler.
+ */
+void at_set_character_handler(struct at *at, at_character_handler_t handler);
```

Comment:
This will need an implementation as well. In the original tree it was added in `at-freertos.c` but I don't think it needs to be a platform specific version.